### DISCUSSION TITLE
How well supported is the odroid M1?

### DISCUSSION THREAD

**Discussion by aaronouthier** [blakeblackshear/frigate#4590] [General]
Hello,
I have 7 IP Cameras - 5 connected, 2 more awaiting mounting. Will likely add more eventually. I’m currently using motioneye on an XU4 with a CloudShell2 add-on. It runs ok. Just heard about Frigate, and am quite excited to try it.

I am considering getting an odroid M1.

Questions include:
Does Frigate even run on ARM platforms?
Would the built-in NPU on the M1 be enough to handle 7-12 cameras in frigate?
If I later decide to add a coral NPU, would one of the M.2 modules work in the slot on my M1, or is that slot for storage only?

For what it’s worth: I plan to boot via eMMC, and store videos on a SATA SSD.

**Comment by NickM-27** [comment]
The rockchip NPU is not supported for detection, frigate does run on arm as a considerable amount of users run on the pi

**Reply by aaronouthier** [reply]
Good to know!

Is there a major technical reason for that, or is the M1 simply a new board, and nobody’s taken the time? Ie. is it likely to be supported in the future? If so, is the time frame more likely to be months or years?

I’ll probably try on my Zima board. Need to pickup a coral USB TPU first though. It would be awesome to have it working by New Years! I’m stoked.

**Reply by NickM-27** [reply]
Right now frigate only supports the coral for acceleration. 0.12 will support more detectors but it's a recent effort. Some contributors have expressed interest in trying to get this to work.

**Comment by MaBeniu** [comment]
are there any updates for odroid M1 support?

**Comment by MaBeniu** [comment]
With lots of testing I would say detection part is not that big of a problem as you can use coral to handle it. I bought coral and now most of the Odroid M1 hardware resources used are for decoding the stream and identifying motion (8 cameras using continuously around 75% CPU load).
That is due to missing hardware acceleration support and ffmpeg libraries that does not support rockchip hardware. Unfortunately it is probably low amount of Odroid m1 users which explains lack of progress/ interest.

Even though I think this device would be capable of dealing with such demand I will start looking to other hardware as Home Assistant OS is something I need in addition with Frigate :)
